session.go

```diff
@@ -555,9 +559,24 @@ func (s *session) Execute(sql string) ([]ast.RecordSet, error) {
 		// return the first recordset if client doesn't support ClientMultiResults.
 		rs = rs[:1]
 	}
+	if len(rs) > 0 {
+		rs[len(rs)-1] = wrapRecordSet{rs[len(rs)-1], s}
+	} else if rs == nil {
+		s.setProcessInfo("")
+	}
 	return rs, nil
 }
 
+type wrapRecordSet struct {
```

Comment:
Close() seems to be a proper place to reset the processInfo, so I wrap a struct around the origin RecordSet, and overload it's Close() method to do it.

I also tried do it in CommitTxn or RollbackTxn, but it doesn't work well. The Info field of the results always get null, even the session that handle show processlist query.

Any suggestion here?